Release Version 6.4.0

thomasplevy opened the pull request:
+ Fixes #1465 via #1748
+ Fixes #1685 and Fixes #2052 via #2108
+ Fixes #1938 via #2016
+ Fixes #2058 via #2100
+ Fixes #2070 via #2113
+ Fixes #2092 via #2112
+ Fixes #2098 via #2118
+ Fixes #2111 via #2119
+ #2104
+ #2114
+ #2116
+ #2117
+ #2120

eri-trabiccolo committed:
Only populate effectively used merged data

eri-trabiccolo committed:
Reduce the list of used merge codes to replace without using potentially overkilling pregs.

pondermatic committed:
Fix display of merge codes in reusable blocks on certificate templates.

github-actions[bot] committed:
Merge branch 'trunk' of https://github.com/gocodebox/lifterlms into dev

pondermatic committed:
Allow classes extended from the manual payment gateway class to display payment instructions.

pondermatic committed:
Allow classes extended from the manual payment gateway class to display payment instructions.

eri-trabiccolo committed:
Initial work core forms detection

nrherron92 committed:
Updated terms and conditions block to hold inline-block setting to fix OceanWP issue

eri-trabiccolo committed:
polish

eri-trabiccolo committed:
introduce LLMS_Forms::is_a_core_form()

pondermatic committed:
Improve `load_reusable_blocks()` documentation.

eri-trabiccolo committed:
fix table name

pondermatic committed:
Add strict array type check.

Co-authored-by: Rocco Aliberti <raliberti84@gmail.com>

nrherron92 committed:
adding changelog

eri-trabiccolo committed:
add tests

thomasplevy committed:
Merge pull request #2104 from gocodebox/custom-manual-payment-gateway

Display instructions in extended manual payment gateway classes

eri-trabiccolo committed:
changelog

github-actions[bot] committed:
Merge branch 'trunk' of https://github.com/gocodebox/lifterlms into dev

thomasplevy committed:
Merge pull request #2106 from nrherron92/issue/1938

Updated terms and conditions block to hold inline-block setting to fi…

eri-trabiccolo committed:
Apply suggestions from code review

Co-authored-by: James Richards <5377968+pondermatic@users.noreply.github.com>

eri-trabiccolo committed:
update tests with review suggestion

eri-trabiccolo committed:
Apply suggestions from code review

Co-authored-by: James Richards <5377968+pondermatic@users.noreply.github.com>

thomasplevy committed:
Merge pull request #2108 from eri-trabiccolo/issue/2052

Implement core form detection that excludes duplicates.

eri-trabiccolo committed:
add tests

pondermatic committed:
Move replacement of reusable blocks to LLMS_Abstract_User_Engagement::merge_content().

pondermatic committed:
Add testing of awarding and syncing with a reusable block.

pondermatic committed:
Remove incorrect changelog entry.

eri-trabiccolo committed:
Force no cache on quiz pages

eri-trabiccolo committed:
Prevent quiz starting (ajax) when attempts limit reached

eri-trabiccolo committed:
Prevent answering a quiz question when attempts limit reached

eri-trabiccolo committed:
changelogs

pondermatic committed:
Initialize shortcodes before the WP 'init' hook calls other LifterLMS methods/functions.

eri-trabiccolo committed:
Apply suggestions from code review

Co-authored-by: Thomas Patrick Levy <thomas@gocodebox.com>

eri-trabiccolo committed:
rework additional_nocache_headers

eri-trabiccolo committed:
wording

pondermatic committed:
Allow filtering of the `LLMS_Shortcode_User_Info` class
by the `llms_load_shortcodes` and `llms_load_shortcode_path` hooks.

eri-trabiccolo committed:
Update .changelogs/issue_1465.yml

Co-authored-by: Thomas Patrick Levy <thomas@gocodebox.com>

pondermatic committed:
Mark incomplete tests.

thomasplevy committed:
Merge pull request #2115 from gocodebox/risky-tests

Mark incomplete tests.

eri-trabiccolo committed:
Apply suggestions from code review

Co-authored-by: James Richards <5377968+pondermatic@users.noreply.github.com>

eri-trabiccolo committed:
doc

eri-trabiccolo committed:
Update includes/abstracts/llms.abstract.notification.view.php

Co-authored-by: James Richards <5377968+pondermatic@users.noreply.github.com>

pondermatic committed:
Moved LLMS_Shortcodes instantiation.

thomasplevy committed:
Use `llms_redirect_and_exit()` and add tests

thomasplevy committed:
Introduce 'automatic' masking of secure/sensitive strings in gateway logs

thomasplevy committed:
add phpcs ignores

eri-trabiccolo committed:
Apply code review suggestions

eri-trabiccolo committed:
Added tests for LLMS_Cache_Helper::additional_nocache_headers() method

eri-trabiccolo committed:
Added tests for `LLMS_Test_LLMS_Student_Quizzes::get_attempts_remaining_for_quiz()` method

eri-trabiccolo committed:
dot

eri-trabiccolo committed:
ajax quiz handler test file

eri-trabiccolo committed:
Add tests on quiz start/answer question for which regards to the attempts limit

thomasplevy committed:
Merge pull request #2116 from gocodebox/man-gateway-test

Manual gateway: Use `llms_redirect_and_exit()` and add tests

eri-trabiccolo committed:
Added Aosta as Italian province

eri-trabiccolo committed:
changelogs

thomasplevy committed:
Merge pull request #1748 from eri-trabiccolo/issue/1465

Notifications: Only populate effectively used merged data

thomasplevy committed:
Merge pull request #2114 from gocodebox/filter-LLMS_Shortcode_User_Info

Allow filtering of the `LLMS_Shortcode_User_Info` class

thomasplevy committed:
Merge pull request #2118 from eri-trabiccolo/issue/2098

Issue/2098

thomasplevy committed:
Merge branch 'dev' of github.com:gocodebox/lifterlms into gateway-masked-logging

eri-trabiccolo committed:
pall-mall is better than golf

thomasplevy committed:
Don't transform filter input

thomasplevy committed:
Apply suggestions from code review

Co-authored-by: Rocco Aliberti <raliberti84@gmail.com>

thomasplevy committed:
Use ternary condition

thomasplevy committed:
Fix cs

thomasplevy committed:
Merge branch 'gateway-masked-logging' of github.com:gocodebox/lifterlms into gateway-masked-logging

eri-trabiccolo commented on includes/forms/class-llms-forms.php:
```diff
@@ -630,6 +630,80 @@ public function get_form_post( $location, $args = array() ) {
+	 * @param string $return What to return: 'posts', for an array of WP_Post; 'ids' for an array of WP_Post ids.
+	 * @return WP_Post[]|int[]
+	 */
+	private function get_core_forms( $return = 'posts', $use_cache = true ) {
```
hey cc people here don't let me play golf!

thomasplevy replied:
Sometimes you play golf like an idiot though, so...

eri-trabiccolo replied:
Ahah, lovely 🌹

thomasplevy committed:
Update includes/functions/llms.functions.log.php

Co-authored-by: Rocco Aliberti <raliberti84@gmail.com>

thomasplevy committed:
changelogs

thomasplevy committed:
Merge branch 'gateway-masked-logging' of github.com:gocodebox/lifterlms into gateway-masked-logging

thomasplevy committed:
Merge pull request #2117 from gocodebox/gateway-masked-logging

Introduce "Secure" string anonymization for payment gateways

pondermatic committed:
Follow existing bootstrap standard.

Co-authored-by: Thomas Patrick Levy <thomasplevy@gmail.com>

thomasplevy committed:
Update includes/shortcodes/class.llms.shortcodes.php

thomasplevy committed:
Merge branch 'dev' into issue/2070-shortcodes-in-engagement-email

thomasplevy committed:
Merge pull request #2113 from gocodebox/issue/2070-shortcodes-in-engagement-email

Initialize shortcodes early

eri-trabiccolo committed:
Load LifterLMS PHP templates at `template_include|10` except the single
restricted content which is loaded at `template_include|100` for compatibility
with theme builders such as Divi and Elementor Pro ones.

eri-trabiccolo committed:
changelogs

eri-trabiccolo committed:
fix tests

eri-trabiccolo committed:
Stop using deprecated `FILTER_SANITIZE_STRING`.

eri-trabiccolo committed:
Update .changelogs/filter_sanitize_string.yml

Co-authored-by: James Richards <5377968+pondermatic@users.noreply.github.com>

thomasplevy committed:
Merge pull request #2120 from eri-trabiccolo/filter_sanitize_string

Stop using deprecated `FILTER_SANITIZE_STRING`.

thomasplevy committed:
Merge pull request #2119 from eri-trabiccolo/issue/2111

Fix compatibility with the Elementor Pro theme builder on courses/archives catalogs

thomasplevy committed:
Merge pull request #2112 from eri-trabiccolo/issue/2092

Force no cache on quiz pages

pondermatic committed:
Add refresh if the edited content contains a WordPress reusable block.

thomasplevy committed:
Merge branch 'dev' into issue/2058-certificate-merge-code-reusable-block

thomasplevy committed:
Merge pull request #2100 from gocodebox/issue/2058-certificate-merge-code-reusable-block

Fix display of merge codes in reusable blocks on certificates.

thomasplevy committed:
Build 6.4.0

thomasplevy committed:
Merge pull request #2125 from gocodebox/build/640

Build/640
